net-misc/icaclient/icaclient-20.12.0.12-r2.ebuild

```diff
@@ -0,0 +1,256 @@
+
+	exeinto "${ICAROOT}"/lib
+	if use amd64 ; then
+		rm lib/ctxjpeg_fb_8.so || die
+	fi
+	doexe lib/*.so
+
+	for dest in "${ICAROOT}"{,/nls/en{,.UTF-8}} ; do
+		insinto "${dest}"
+		doins nls/en.UTF-8/eula.txt
+	done
+
+	insinto "${ICAROOT}"
+	doins -r usb
+
+	insinto "${ICAROOT}"/config
+	# nls/en/*.ini is being handled by pkg_postinst()
+	doins config/* config/.*
+	for tmpl in {appsrv,wfclient}.template ; do
+		newins nls/en/${tmpl} ${tmpl/template/ini}
+	done
+	touch "${ED}/${ICAROOT}"/config/.server || die
+
+	insinto "${ICAROOT}"/gtk
+	doins gtk/*
+
+	insinto "${ICAROOT}"/gtk/glade
+	doins gtk/glade/*
+
+	insinto "${ICAROOT}"/site
+	doins -r site/*
+
+	dodir "${ICAROOT}"/help
+
+	insinto "${ICAROOT}"/config/usertemplate
+	doins config/usertemplate/*
+
+	local lang LANGCODES=( en )
+	use l10n_de && LANGCODES+=( de )
+	use l10n_es && LANGCODES+=( es )
+	use l10n_fr && LANGCODES+=( fr )
+	use l10n_ja && LANGCODES+=( ja )
+	use l10n_zh-CN && LANGCODES+=( zh_CN )
+
+	for lang in ${LANGCODES[@]} ; do
+		insinto "${ICAROOT}"/nls/${lang}
+		doins nls/${lang}/*
+
+		insinto "${ICAROOT}"/nls/$lang/UTF-8
+		doins nls/${lang}.UTF-8/*
+
+		insinto "${ICAROOT}"/nls/${lang}/LC_MESSAGES
+		doins nls/${lang}/LC_MESSAGES/*
+
+		insinto "${ICAROOT}"/nls/${lang}
+		dosym UTF-8 "${ICAROOT}"/nls/${lang}/utf8
+
+		for tmpl in {appsrv,wfclient}.template ; do
+			cp "${ED}/${ICAROOT}"/nls/${lang}/${tmpl} \
+				"${ED}/${ICAROOT}"/nls/${lang}/${tmpl/template/ini} \
+				|| die
+		done
+	done
+
+	insinto "${ICAROOT}"/nls
+	dosym en /opt/Citrix/ICAClient/nls/C
+
+	insinto "${ICAROOT}"/icons
+	doins icons/*
+
+	insinto "${ICAROOT}"/keyboard
+	doins keyboard/*
+
+	rm -r "${S}"/keystore/cacerts || die
+	dosym ../../../../etc/ssl/certs "${ICAROOT}"/keystore/cacerts
+
+	local util_files=(
+		configmgr
+		conncenter
+		ctx_app_bind
+		ctx_rehash
+		ctxlogd
+		ctxwebhelper
+		gst_play1.0
+		gst_read1.0
+		hdxcheck.sh
+		icalicense.sh
+		libgstflatstm1.0.so
+		lurdump
+		new_store
+		nslaunch
+		pnabrowse
+		setlog
+		storebrowse
+		sunraymac.sh
+		webcontainer
+		what
+		xcapture
+	)
+
+	exeinto "${ICAROOT}"/util
+	for bin in ${util_files[@]} ; do
+		doexe util/${bin}
+	done
+
+	# https://bugs.gentoo.org/655922
+	dosym gst_play1.0 "${ICAROOT}"/util/gst_play
+	dosym gst_read1.0 "${ICAROOT}"/util/gst_read
+	dosym libgstflatstm1.0.so "${ICAROOT}"/util/libgstflatstm.so
+
+	doenvd "${FILESDIR}"/10ICAClient
+
+	for bin in configmgr conncenter new_store ; do
+		make_wrapper ${bin} "${ICAROOT}"/util/${bin} . "${ICAROOT}"/util
+	done
+
+	for bin in selfservice wfica ; do
+		make_wrapper ${bin} "${ICAROOT}"/${bin} . "${ICAROOT}"
+	done
+
+	dodir /etc/revdep-rebuild/
+	echo "SEARCH_DIRS_MASK=\"${ICAROOT}\"" \
+		> "${ED}"/etc/revdep-rebuild/70icaclient
+
+	ICAPORT="linuxx64"
+	if use x86; then
+		ICAPORT="linuxx86"
+	fi
+	dodir "${ICAROOT}"/pkginf
+	insinto "${ICAROOT}"/pkginf
+	newins "${WORKDIR}"/PkgId Ver.core."${ICAPORT}"
```

Comment:
Instead of that newly introduced `ICAPORT` variable did you try to use the `ICAARCH` variable that is also being used in `pkg_setup()`? That would avoid duplicating code that is already present.